### PR TITLE
IJ samples PR init

### PR DESCRIPTION
**Pull request by shouryaj**
# PR into Azure/azureml-examples
WIP not to be merged.

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes
Adding samples for Interactive Jobs
-

fixes #

### CODE REVIEW

File: cli/jobs/basics/hello-interactive.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+command: echo "hello world" | sleep 3600
+environment:
+  image: azureml:AzureML-sklearn-1.0-ubuntu20.04-py38-cpu@latest
+compute: azureml:cpu-cluster
+services: 
+  my_vscode: 
+    job_service_type: VS_code
```
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:8]: lower case "vs_code".

File: cli/jobs/basics/hello-interactive.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+environment:
+  image: azureml:AzureML-sklearn-1.0-ubuntu20.04-py38-cpu@latest
+compute: azureml:cpu-cluster
+services: 
+  my_vscode: 
+    job_service_type: VS_code 
+  my_jupyter_lab: 
+    job_service_type: jupyter_lab 
+  my_tensorboard:
+    job_service_type: tensorboard
```
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:12]: probably in comments add how to do SSH here.

File: cli/jobs/basics/hello-interactive.yml (hunk 1 above)
Comment (TonyJ1) [reply cli/jobs/basics/hello-interactive.yml:8]: yes, the values are case sensitive and should be one of the below
"jupyter_lab", "ssh", "tensor_board", "vs_code"

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py#:~:text=job_service_type%3A%20Optional%5BLiteral%5B%22jupyter_lab%22%2C%20%22ssh%22%2C%20%22tensor_board%22%2C%20%22vs_code%22%5D%5D%20%3D%20None%2C

File: cli/jobs/basics/hello-interactive.yml (hunk 2 above)
Comment (TonyJ1) [reply cli/jobs/basics/hello-interactive.yml:12]: "tensor_board"

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb [hunk 3]
```diff
@@ -60,9 +70,9 @@
    "outputs": [],
    "source": [
     "# Enter details of your AML workspace\n",
-    "subscription_id = \"<SUBSCRIPTION_ID>\"\n",
-    "resource_group = \"<RESOURCE_GROUP>\"\n",
-    "workspace = \"<AML_WORKSPACE_NAME>\""
+    "subscription_id = \"381b38e9-9840-4719-a5a0-61d9585e1e91\"\n",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:73]: remove the actual values

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb [hunk 4]
```diff
@@ -179,9 +281,9 @@
    "name": "python3-azureml"
   },
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
+   "display_name": "Python 3.10.6 ('v2_final')",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:284]: may want to keep the spec as before

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb [hunk 5]
```diff
@@ -193,10 +295,15 @@
...
-   "version": "3.7.12"
+   "version": "3.10.6"
   },
   "nteract": {
    "version": "nteract-front-end@1.0.0"
+  },
+  "vscode": {
+   "interpreter": {
+    "hash": "2346ebb0d49c319bb50d0e319d661980ccd11854fd8dafc39695d4051b27368b"
+   }
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:306]: delete this?

Comment (shouryaj) [reply sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb]: done

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb (hunk 4 above)
Comment (shouryaj) [reply sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb]: done

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb (hunk 3 above)
Comment (shouryaj) [reply sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb]: done

File: cli/jobs/basics/hello-interactive.yml (hunk 2 above)
Comment (shouryaj) [reply cli/jobs/basics/hello-interactive.yml]: fixed + added ssh in comment

File: cli/jobs/basics/hello-interactive.yml (hunk 1 above)
Comment (shouryaj) [reply cli/jobs/basics/hello-interactive.yml]: done

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb [hunk 6]
```diff
@@ -193,7 +213,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.12"
+   "version": "3.10.6"
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:216]: why this change? should we revert?

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb [hunk 7]
```diff
@@ -179,9 +199,9 @@
    "name": "python3-azureml"
   },
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
+   "display_name": "Python 3.10",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:202]: should we revert this change?

File: cli/jobs/basics/hello-interactive.yml [hunk 8]
```diff
@@ -0,0 +1,24 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+command: echo "hello world" | sleep 3600
```
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:2]: Can you reduce the sleep time maybe?

File: cli/jobs/pipelines-with-components/basics/1b_e2e_registered_components/pipeline.yml [hunk 9]
```diff
@@ -32,6 +32,24 @@ jobs:
       learning_rate_schedule: ${{parent.inputs.pipeline_job_learning_rate_schedule}}
     outputs:
       model_output: ${{parent.outputs.pipeline_job_trained_model}}
```
Comment (srsaggam) [inline cli/jobs/pipelines-with-components/basics/1b_e2e_registered_components/pipeline.yml:34]: Do you want to create a new sample? How users can discover this example?

File: cli/jobs/basics/hello-interactive.yml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  my_vscode: 
+    job_service_type: vs_code
+    nodes: all
+  my_jupyter_lab: 
+    job_service_type: jupyter_lab
+    nodes: all
+  my_tensorboard:
+    job_service_type: tensor_board
+    properties:
+      logDir: "logs/tblogs"
```
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:18]: change to outputs/tblogs

File: cli/jobs/basics/hello-interactive.yml (hunk 8 above)
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:2]:
```suggestion
command: echo "hello world" | sleep 900
```

File: cli/jobs/basics/hello-interactive.yml (hunk 10 above)
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:18]:
```suggestion
      logDir: "outputs/tblogs"
```

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb (hunk 7 above)
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:202]:
```suggestion
   "display_name": "Python 3.10 - SDK V2",
```

File: sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb [hunk 11]
```diff
@@ -179,9 +199,9 @@
...
    "language": "python",
-   "name": "python310-sdkv2"
+   "name": "python3"
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/tensorflow/mnist/tensorflow-mnist.ipynb:204]:
```suggestion
   "name": "python310-sdkv2"
```

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 12]
```diff
@@ -244,9 +261,9 @@
    "description": "Create a pipeline with command function"
   },
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
+   "display_name": "Python 3.10.6 ('v2_final')",
```
Comment (srsaggam) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:264]:
```suggestion
   "display_name": "Python 3.10 - SDK V2",
```

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 13]
```diff
@@ -244,9 +261,9 @@
...
    "language": "python",
-   "name": "python310-sdkv2"
+   "name": "python3"
```
Comment (srsaggam) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:266]:
```suggestion
   "name": "python310-sdkv2"
```

File: cli/jobs/pipelines-with-components/basics/1b_e2e_registered_components/pipeline.yml [hunk 14]
```diff
@@ -32,6 +32,24 @@ jobs:
...
+      my_vscode: 
+        job_service_type: vs_code
+        nodes: all
+      my_jupyter_lab: 
+        job_service_type: jupyter_lab
+        nodes: all
+      my_tensorboard:
+        job_service_type: tensor_board
+        properties:
+          logDir: "logs/tblogs"
```
Comment (srsaggam) [inline cli/jobs/pipelines-with-components/basics/1b_e2e_registered_components/pipeline.yml:45]:
```suggestion
          logDir: "outputs/tblogs"
```

File: cli/jobs/basics/hello-interactive.yml [hunk 15]
```diff
@@ -0,0 +1,20 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+command: python hello-interactive.py && sleep 900
```
Comment (srsaggam) [inline cli/jobs/basics/hello-interactive.yml:2]:
```suggestion
command: python hello-interactive.py && sleep 600
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 16]
```diff
@@ -228,9 +238,9 @@
    "name": "python3-azureml"
   },
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
+   "display_name": "Python (myenv)",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:241]:
```suggestion
   "display_name": "Python 3.10 - SDK V2",
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 17]
```diff
@@ -228,9 +238,9 @@
...
    "language": "python",
-   "name": "python310-sdkv2"
+   "name": "myenv"
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:243]:
```suggestion
   "name": "python310-sdkv2"
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 18]
```diff
@@ -242,7 +252,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.12"
+   "version": "3.9.13"
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:255]:
```suggestion
   "version": "3.7.12"
```

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 19]
```diff
@@ -244,9 +254,9 @@
    "description": "Create a pipeline with command function"
   },
   "kernelspec": {
-   "display_name": "Python 3.10 - SDK V2",
+   "display_name": "Python (myenv)",
```
Comment (srsaggam) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:257]:
```suggestion
   "display_name": "Python 3.10 - SDK V2",
```

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 20]
```diff
@@ -244,9 +254,9 @@
...
    "language": "python",
-   "name": "python310-sdkv2"
+   "name": "myenv"
```
Comment (srsaggam) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:259]:
```suggestion
   "name": "python310-sdkv2"
```

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 21]
```diff
@@ -258,7 +268,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.12"
+   "version": "3.9.13"
```
Comment (srsaggam) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:271]:
```suggestion
   "version": "3.8.12"
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 22]
```diff
@@ -60,7 +61,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 2,
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:64]:
```suggestion
   "execution_count": null,
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 23]
```diff
@@ -79,14 +80,14 @@
    "outputs": [],
    "source": [
     "# Enter details of your AML workspace\n",
-    "subscription_id = \"<SUBSCRIPTION_ID>\"\n",
-    "resource_group = \"<RESOURCE_GROUP>\"\n",
-    "workspace = \"<AML_WORKSPACE_NAME>\""
+    "subscription_id = \"381b38e9-9840-4719-a5a0-61d9585e1e91\"\n",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:83]:
```suggestion
    "subscription_id = \"<SUBSCRIPTION_ID>\"\n",
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 24]
```diff
@@ -79,14 +80,14 @@
...
+    "resource_group = \"sasik_rg\"\n",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:84]:
```suggestion
    "resource_group = \"<RESOURCE_GROUP>\"\n",
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 25]
```diff
@@ -79,14 +80,14 @@
...
+    "workspace = \"sasik_canary\""
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:85]:
```suggestion
    "workspace = \"<AML_WORKSPACE_NAME>\""
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 26]
```diff
@@ -151,7 +152,19 @@
...
...\",\u001b[39;00m\n\u001b[0;32m     19\u001b[0m         \u001b[38;5;66;03m# properties={\u001b[39;00m\n\u001b[0;32m     20\u001b[0m         \u001b[38;5;66;03m#   \"sshPublicKeys\":\"<add-public-key>\"\u001b[39;00m\n\u001b[0;32m     21\u001b[0m         \u001b[38;5;66;03m# }\u001b[39;00m\n\u001b[0;32m     22\u001b[0m         \u001b[38;5;66;03m# ) # uncomment add SSH Public Key to access job container via SSH\u001b[39;00m\n\u001b[0;32m     23\u001b[0m     },\n\u001b[0;32m     24\u001b[0m     environment\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAzureML-sklearn-0.24-ubuntu18.04-py37-cpu@latest\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m     25\u001b[0m     compute\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcpu-cluster\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m     26\u001b[0m     display_name\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msklearn-iris-example\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m     27\u001b[0m     \u001b[38;5;66;03m# experiment_name\u001b[39;00m\n\u001b[0;32m     28\u001b[0m     \u001b[38;5;66;03m# description\u001b[39;00m\n\u001b[0;32m     29\u001b[0m )\n",
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:163]:
```suggestion
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 27]
```diff
@@ -151,7 +152,19 @@
...
+      "\u001b[1;31mNameError\u001b[0m: name 'JobService' is not defined"
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:164]:
```suggestion
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 28]
```diff
@@ -151,7 +152,19 @@
...
+     ]
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:165]:
```suggestion
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 29]
```diff
@@ -151,7 +152,19 @@
...
+    }
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:166]:
```suggestion
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 30]
```diff
@@ -151,7 +152,19 @@
...
+   ],
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:167]:
```suggestion
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 31]
```diff
@@ -86,7 +87,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 3,
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:90]:
```suggestion
   "execution_count": null,
```

File: sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb [hunk 32]
```diff
@@ -135,7 +136,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 4,
```
Comment (srsaggam) [inline sdk/python/jobs/single-step/scikit-learn/iris/iris-scikit-learn.ipynb:139]:
```suggestion
   "execution_count": null,
```

File: cli/jobs/pipelines-with-components/basics/1b_e2e_registered_components/pipeline.yml [hunk 33]
```diff
@@ -32,6 +32,21 @@ jobs:
...
+      my_tensorboard:
+        job_service_type: tensor_board
+        properties:
+          logDir: "outputs/tblogs"
+    #  my_ssh:
+    #    job_service_type: tensor_board
+    #    properties:
+    #      sshPublicKeys: <paste the entire pub key content>
+    #    nodes: all # Use the `nodes` property to pick which node you want to enable interactive services on. If `nodes` are not selected, by default, interactive applications are only enabled on the head node.
+
```
Comment (srsaggam) [inline cli/jobs/pipelines-with-components/basics/1b_e2e_registered_components/pipeline.yml:49]:
```suggestion
```

File: cli/jobs/basics/src/hello-interactive.py [hunk 34]
```diff
@@ -0,0 +1,57 @@
+from datetime import datetime
+import tensorflow as tf
+from tensorflow import keras
+import time
+import numpy as np
+
```
Comment (srsaggam) [inline cli/jobs/basics/src/hello-interactive.py:6]:
```suggestion
```

**Review by srsaggam — APPROVED** [review]

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 35]
```diff
@@ -143,6 +143,16 @@
     "        file_name=file_name,\n",
     "    ),\n",
     "    outputs={},\n",
+    "    services={\n",
+    "        \"Jupyterlab endpoint\": JobService(job_service_type=\"jupyter_lab\"),\n",
+    "        \"Vscode endpoint\": JobService(job_service_type=\"vs_code\"),\n",
+    "        # \"My_ssh\": JobService(\n",
+    "        # job_service_type = \"ssh\",\n",
+    "        # properties={\n",
+    "        #   \"sshPublicKeys\":\"<add-public-key>\"\n",
```
Comment (harneetvirk) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:152]: How is the public key replaced/updated on the automated validations?

Comment (srsaggam) [reply sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:152]: It is a limitation in current preview. But we have a feature where this scenario will be possible. Basically, it will allow user to put a new key in the remote machine on the fly.

**Review by jeff-shepherd — APPROVED** [review]

**Review by TonyJ1 — APPROVED** [review]
